Add the gdal retile algorithm (#4715)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst

     - The HTML file containing information about the input raster layer

.. _gdalretile:

Retile
------

> This utility will retile a set of input tile(s).

Reviewer: ```suggestion
Retiles a set of input tiles.
```
If it's a set then there are many, aren't they?

Author: Cut and paste from GDAL docs...

> All the input tile(s) must be georeferenced in the same

Reviewer: ```suggestion
All the input tiles must be georeferenced in the same
```

> coordinate system and have a matching number of bands.

Reviewer: Kind of formulation I was suggesting for the raster calc

Author: Cut and paste...

Optionally pyramid levels are generated.

> It is possible to generate shape file(s) for the tiled

Reviewer: I can't see a parameter/output that addresses this. Do I miss anything?

Author: Same here, but it is cut and paste.

Reviewer: Yeah, I guessed it was copy and paste (we don't need to reinvent the wheel) but if we don't expose all the features of the original alg, we shouldn't claim that. Even if that can be done with command line options, I'd prefer we stick to what we provide to users, up to advanced gdal techies to do what they know. This is what we usually do with the gdal based algs.

Author: Good point.  I have tried to address this in PR #4724.

output.

This algorithm is derived from the

> `GDAL Retile utility <https://gdal.org/gdal_retile.htmll>`_ .

Reviewer: ```suggestion
`GDAL Retile utility <https://gdal.org/gdal_retile.html>`_ .
```

Parameters
..........

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :stub-columns: 0

   * - Label
     - Name
     - Type
     - Description
   * - **Input files**
     - ``INPUT``
     - [raster] [list]

>      - The input raster files (can be multiple files).

Reviewer: Do we need the comments in parenthesis (here and the next two)? Should be obvious.

Author: No.  Some kind of modified cut and paste, I think.

>      - The input raster files (can be multiple files).

Author: ```suggestion
     - The input raster files
```

   * - **Tile width**
     - ``TILE_SIZE_X``
     - [number]

       Default: 256

>      - Width of the tiles in pixels (minimum 0)

Reviewer: 0 is allowed?

Author: So it says.

   * - **Tile height**
     - ``TILE_SIZE_Y``
     - [number]

       Default: 256
     - Height of the tiles in pixels (minimum 0)

>    * - **Overlap in pixels between consequtive tiles**

Reviewer: ```suggestion
   * - **Overlap in pixels between consecutive tiles**
```

     - ``OVERLAP``
     - [number]

       Default: 0
     - 
   * - **Number of pyramid levels to build**
     - ``LEVELS``
     - [number]

       Default: 1
     - Minimum: 0
   * - **Source coordinate reference system**
     - ``SOURCE_CRS``
     - [crs]

       Default: None
     - 
   * - **Resampling method**
     - ``RESAMPLING``
     - [enumeration]

       Default: 0
     - The resampling algorithm to be used

       Options:

       * 0 --- Nearest Neighbour (``nearest``)
       * 1 --- Bilinear (``bilinear``)
       * 2 --- Cubic (``cubic``)
       * 3 --- Cubic Spline (``cubicspline``)
       * 4 --- Lanczos Windowed Sinc (``lanczos``)

   * - **Column delimiter used in the CSV file**

       Optional
     - ``DELIMITER``
     - [string]

       Default: ';'

>      - CSV file delimiter

Reviewer: I wondered what a csv file would be helpful for here, so maybe
```suggestion
     - Delimiter to use in the CSV file containing the tile(s) georeferencing information
```

       Optional
     - ``EXTRA``
     - [string]

       Default: ''
     - Add extra GDAL command line options
   * - **Output data type**
     - ``DATA_TYPE``
     - [enumeration]
       
       Default: 5
     - Defines the format of the output raster file.

       Options:

       * 0 --- Byte
       * 1 --- Int16
       * 2 --- UInt16
       * 3 --- UInt32
       * 4 --- Int32
       * 5 --- Float32
       * 6 --- Float64
       * 7 --- CInt16
       * 8 --- CInt32
       * 9 --- CFloat32
       * 10 --- CFloat64

   * - **Build only the pyramids**
     - ``ONLY_PYRAMIDS``
     - [boolean]

       Default: False
     - 
   * - **Use separate directory for each tile row**
     - ``DIR_FOR_ROW``
     - [boolean]

       Default: False
     - 
   * - **Output directory**
     - ``OUTPUT``
     - [folder]

       Default: ``[Save to temporary folder]``
     - Specify the output folder for the tiles.
       One of:

       * Save to Temporary Directory
       * Save to Directory...

       The file encoding can also be changed here.
   * - **CSV file containing the tile(s) georeferencing information**
     - ``OUTPUT_CSV``
     - [file]

       Default: ``[Skip output]``

>      - Specify the output folder for the tiles.

Reviewer: No no...

Author: :-)

>      - Specify the output folder for the tiles.

Author: ```suggestion
     - 
```

       One of:

       * Skip Output
       * Save to a Temporary File
       * Save to File...

       The file encoding can also be changed here.

Outputs
.......

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :stub-columns: 0

   * - Label
     - Name
     - Type
     - Description
   * - **Output directory**
     - ``OUTPUT``
     - [folder]
     - The output folder for the tiles.
   * - **CSV file containing the tile(s) georeferencing information**
     - ``OUTPUT_CSV``
     - [file]

>      - The output folder for the tiles.

Reviewer: No no...

>      - The output folder for the tiles.

Author: ```suggestion
     - The CSV file with georeferencing information for the tiles.
```
